scripts/fuzz_opt.py

```diff
@@ -185,9 +288,34 @@ def handle_pair(self, input, before_wasm, after_wasm, opts):
     def can_run_on_feature_opts(self, feature_opts):
         return True
 
+    def increment_runs(self):
+        self.num_runs += 1
+
+    def count_runs(self):
+        return self.num_runs
+
 
 # Run VMs and compare results
+
+class VM:
+    def __init__(self, name, run, deterministic_nans):
+        self.name = name
+        self.run = run
+        self.deterministic_nans = deterministic_nans
+
+
 class CompareVMs(TestCaseHandler):
+    def __init__(self):
+        super(CompareVMs, self).__init__()
+        self.vms = [
+            VM('binaryen interpreter', lambda js, wasm: run_bynterp(wasm, ['--fuzz-exec-before']),               deterministic_nans=True), # noqa
+            VM('d8',                   lambda js, wasm: run_vm([shared.V8, js] + shared.V8_OPTS + ['--', wasm]), deterministic_nans=False), # noqa
```

Comment:
I think this would be more readable (and not require `#noqa`) with the lambdas extracted into named inner functions.